lib/spack/spack/test/llnl/util/link_tree.py

```diff
@@ -133,6 +194,41 @@ def test_merge_to_existing_directory(stage, link_tree):
         assert not os.path.isfile("dest/c/d/e/7")
 
 
+@pytest.mark.skipif(sys.platform != "win32", reason="Windows only.")
+def test_merge_to_existing_directory__win32_base(stage, link_tree, monkeypatch):
```

Comment:
The tests are not exactly the same. the `check_file_link()` call in the "original" unit test does not work on base-priv windows.

Can you maybe explain in a little more detail exactly what you want to be parameterized becauseI don't fully understand what you are asking for. 

I can add something like `@pytest.mark.parametrize('win_can_symlink', [True, False])` to the `test_merge_to_existing_directory` test, but then it has to run twice on every OS. And it would fail if `_win_can_symlink()` is _actually_ `False` and we set it to be `True`.

Reply:
You could:

* instead of monkeypatching when `win_can_symlink` is `True` and we can't actually symlink, just skip (you can add a skip call inside of a test)
* Perform the `check_file_link` only when `win_can_symlink` is `True`

At some point the number of accommodations for the parameterization outweighs the reduction in duplication, but here I think it's worthwhile. Let me know if I overlooked something though (other spots that would complicate merging the two methods).

Reply:
Ok I changed these. I still had to monkeypatch `_windows_can_symlink` because the tests need to run both as base and dev-mode.